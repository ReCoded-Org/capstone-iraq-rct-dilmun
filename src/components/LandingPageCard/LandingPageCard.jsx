import React from 'react'
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Image from '../../assets/profile_bg.png'

export default function LandingPageCard({ productName, description, location, date, views, productImage, price, state}) {
  return (
    <div className="w-72 h-96 bg-white rounded-3xl relative">
      <dev className="relative">
        <img
          src={productImage}
          alt="item card"
          className="bg-cover bg-center rounded-t-3xl"
        />
        <div className="bg-red px-3 py-1 absolute top-44 left-3 rounded-md text-sm">
          {price > 0 ? price : state}
        </div>
      </dev>
      <div className="px-2">
        <div className="text-2xl mt-5">{productName}</div>
        <div className="text-xs">
          {description}
        </div>
        <div className="text-xs flex justify-between mx-3 absolute inset-x-0 bottom-2">
          <div className="flex space-x-1">
            <FontAwesomeIcon icon="eye" />
            <p className="">{views}</p>
          </div>
          <div className="flex space-x-1">
            <FontAwesomeIcon icon="clock" />
            <p>{date.slice(0, 10)}</p>
            <FontAwesomeIcon icon="map-marker-alt" />
            <p>{location}</p>
          </div>
        </div>
      </div>
    </div>
  )
}

LandingPageCard.propTypes = {
  productName: PropTypes.string,
  description: PropTypes.string,
  date: PropTypes.string,
  views: PropTypes.number,
  price: PropTypes.number,
  state: PropTypes.string,
  productImage: PropTypes.string,
  location: PropTypes.string
}

LandingPageCard.defaultProps = {
  productName: '',
  description: '',
  date: '',
  views: 0,
  price: 0,
  state: 'unknown',
  productImage: Image,
  location: ''
}