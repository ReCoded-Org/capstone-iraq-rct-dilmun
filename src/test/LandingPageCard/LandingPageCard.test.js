import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as  Router} from 'react-router-dom'
import LandingPageCard from '../../components/LandingPageCard';

it ("Footer renders correctly", () => {
  const tree = renderer.create(<Router> <LandingPageCard /> </Router>).toJSON();
  expect(tree).toMatchSnapshot();
});