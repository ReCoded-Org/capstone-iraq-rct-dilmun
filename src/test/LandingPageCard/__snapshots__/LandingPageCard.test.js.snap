// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Landing Page Card Passed 1`] = `
Array [
  " ",
  <div
    className="w-72 h-96 bg-white  cursor-pointer rounded-3xl focus:outline-none relative shadow-md hover:shadow-xl transform hover:scale-105 transition duration-300 ease-in-out"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
  >
    <dev
      className="relative"
    >
      <img
        alt="item card"
        className="rounded-t-3xl border  object-cover  h-52 w-full"
        src="noPhoto.jpg"
      />
      <div
        className="bg-red text-white px-3 py-1 absolute top-44 left-3 rounded-md text-sm"
      >
        unknown
      </div>
    </dev>
    <div
      className="px-3 grid"
    >
      <div
        className="text-2xl mt-5 truncate"
      >
        
      </div>
      <p
        className="text-xs h-24 overflow-y-hidden text-center "
      >
        
      </p>
      <div
        className="text-xs flex justify-between px-4 absolute inset-x-0 bottom-2 "
      >
        <div
          className="flex space-x-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-eye fa-w-18 "
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <p
            className=""
          >
            0
          </p>
        </div>
        <div
          className="flex space-x-5 "
        >
          <div
            className="flex space-x-1"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-clock fa-w-16 "
              data-icon="clock"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              
            </p>
          </div>
          <div
            className="flex space-x-1"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-map-marker-alt fa-w-12 "
              data-icon="map-marker-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
